core/templates/components/rubrics-editor/rubrics-editor.component.ts

```diff
@@ -95,6 +98,34 @@ export class RubricsEditorComponent {
     return Boolean(this.editableExplanations[difficulty][index]);
   }
 
+  isExplanationLengthValid(difficulty: string, idx: number): boolean {
+    return this.editableExplanations[difficulty][idx].length <=
+    this.maximumCharacterLengthOfExplanation;
+  }
+
+  isMediumLevelExplanationValid(): boolean {
+    // Checking if medium level rubrics have at least one explantion.
+    return (
+      this.rubrics[this.MEDIUM_EXPLANATION_INDEX] &&
+      this.rubrics[this.MEDIUM_EXPLANATION_INDEX].getExplanations().length >=
+      1);
+  }
+
+  hasReachedExplanationCountLimit(): boolean {
+    let totalExplanations: number = 0;
+    let easyDifficulty = 'Easy';
+    let mediumDifficulty = 'Medium';
+    let hardDifficulty = 'Hard';
+
+    let totalEasyExplanations = this.editableExplanations[easyDifficulty].length;
+    let totalMediumExplanations = this.editableExplanations[mediumDifficulty].length;
+    let totalHardExplanations = this.editableExplanations[hardDifficulty].length;
+
+    totalExplanations = totalEasyExplanations + totalMediumExplanations +
+    totalHardExplanations;
+    return (totalExplanations >= this.maximumNumberofExplanations);
+  }
```

Comment:
I'm not sure why the frontend test is saying that `this.editableExplanations` is `null`, but as a backup, what if we try writing it in this way:
```suggestion
  hasReachedExplanationCountLimit(): boolean {
    let totalExplanations: number = 0;
    for (let difficulty in editableExplanations) {
      if (editableExplanations[difficulty]) {
        totalExplanations += editableExplanations[difficulty].length;
      }
    }
    
    return totalExplanations >= this.maximumNumberofExplanations;
  }
```